File: frontend/src/services/ant-design-pro/platformAiAccountingController.ts
// @ts-ignore
/* eslint-disable */
import { request } from "@umijs/max";

/** 此处后端没有提供注释 GET /api/platform/ai/executions/${param0}/accounting */
export async function accountingDetail(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.accountingDetailParams,
  options?: { [key: string]: any }
) {
  const { executionId: param0, ...queryParams } = params;
  return request<API.ApiResponsePlatformAiAccountingDetailResponse>(
    `/api/platform/ai/executions/${param0}/accounting`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/platform/ai/models/${param0}/billing */
export async function billingHistory(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.billingHistoryParams,
  options?: { [key: string]: any }
) {
  const { modelId: param0, ...queryParams } = params;
  return request<API.ApiResponseModelBillingHistoryResponse>(
    `/api/platform/ai/models/${param0}/billing`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/models/${param0}/cost-price-versions/${param1}/revoke */
export async function revokeCostPrice(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.revokeCostPriceParams,
  options?: { [key: string]: any }
) {
  const { modelId: param0, versionId: param1, ...queryParams } = params;
  return request<API.ApiResponseModelPriceVersionResponse>(
    `/api/platform/ai/models/${param0}/cost-price-versions/${param1}/revoke`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/models/${param0}/point-price-versions */
export async function publishPointPrice(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.publishPointPriceParams,
  body: API.PublishModelPointPriceRequest,
  options?: { [key: string]: any }
) {
  const { modelId: param0, ...queryParams } = params;
  return request<API.ApiResponseModelPointPriceVersionResponse>(
    `/api/platform/ai/models/${param0}/point-price-versions`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/models/${param0}/point-price-versions/${param1}/revoke */
export async function revokePointPrice(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.revokePointPriceParams,
  options?: { [key: string]: any }
) {
  const { modelId: param0, versionId: param1, ...queryParams } = params;
  return request<API.ApiResponseModelPointPriceVersionResponse>(
    `/api/platform/ai/models/${param0}/point-price-versions/${param1}/revoke`,
    {
      method: "POST",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/platform/ai/models/${param0}/price-versions */
export async function publishModelPrice(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.publishModelPriceParams,
  body: API.PublishModelPriceRequest,
  options?: { [key: string]: any }
) {
  const { modelId: param0, ...queryParams } = params;
  return request<API.ApiResponseModelPriceVersionResponse>(
    `/api/platform/ai/models/${param0}/price-versions`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

File: backend/src/main/java/com/antshorttv/accounting/PlatformAiAccountingController.java
package com.antshorttv.accounting;

import com.antshorttv.common.ApiResponse;
import com.antshorttv.platform.RequirePlatformPermission;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/platform/ai")
public class PlatformAiAccountingController {
    private final PlatformAiAccountingService service;

    public PlatformAiAccountingController(PlatformAiAccountingService service) {
        this.service = service;
    }

    @PostMapping("/models/{modelId}/price-versions")
    @RequirePlatformPermission("PLATFORM_AI_PRICE_PUBLISH")
    public ApiResponse<ModelPriceVersionResponse> publishModelPrice(
        @PathVariable Long modelId,
        @Valid @RequestBody PublishModelPriceRequest request
    ) {
        return ApiResponse.success(service.publishModelPrice(modelId, request));
    }

    @GetMapping("/models/{modelId}/billing")
    @RequirePlatformPermission("PLATFORM_AI_ACCOUNTING_VIEW")
    public ApiResponse<ModelBillingHistoryResponse> billingHistory(@PathVariable Long modelId) {
        return ApiResponse.success(service.billingHistory(modelId));
    }

    @PostMapping("/models/{modelId}/point-price-versions")
    @RequirePlatformPermission("PLATFORM_AI_POINT_PRICE_PUBLISH")
    public ApiResponse<ModelPointPriceVersionResponse> publishPointPrice(
        @PathVariable Long modelId,
        @Valid @RequestBody PublishModelPointPriceRequest request
    ) {
        return ApiResponse.success(service.publishModelPointPrice(modelId, request));
    }

    @PostMapping("/models/{modelId}/cost-price-versions/{versionId}/revoke")
    @RequirePlatformPermission("PLATFORM_AI_PRICE_PUBLISH")
    public ApiResponse<ModelPriceVersionResponse> revokeCostPrice(
        @PathVariable Long modelId, @PathVariable Long versionId
    ) {
        return ApiResponse.success(service.revokeCostPrice(modelId, versionId));
    }

    @PostMapping("/models/{modelId}/point-price-versions/{versionId}/revoke")
    @RequirePlatformPermission("PLATFORM_AI_POINT_PRICE_PUBLISH")
    public ApiResponse<ModelPointPriceVersionResponse> revokePointPrice(
        @PathVariable Long modelId, @PathVariable Long versionId
    ) {
        return ApiResponse.success(service.revokePointPrice(modelId, versionId));
    }

    @GetMapping("/executions/{executionId}/accounting")
    @RequirePlatformPermission("PLATFORM_AI_ACCOUNTING_VIEW")
    public ApiResponse<PlatformAiAccountingDetailResponse> accountingDetail(
        @PathVariable Long executionId
    ) {
        return ApiResponse.success(service.accountingDetail(executionId));
    }
}
